### PR TITLE
Kernel module should prefer functions with nx/kmm prefix

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
replace with kernel version

## Impact
Should no functional change

### CODE REVIEW

**Review by davids5 — APPROVED** [review]
LGTM

`Except for the Error: mmap/fs_mmap.c:154:12: error: format '%d' expects argument of type 'int', but argument 3 has type 'size_t' {aka 'long unsigned int'} [-Werror=format=]
       ferr("ERROR: Invalid length, length=%d\n", length);`

**Review by jerpelea — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
> LGTM
> 
> `Except for the Error: mmap/fs_mmap.c:154:12: error: format '%d' expects argument of type 'int', but argument 3 has type 'size_t' {aka 'long unsigned int'} [-Werror=format=] ferr("ERROR: Invalid length, length=%d\n", length);`

@davids5 done, thanks.

File: arch/xtensa/src/esp32/esp32_wifi_adapter.c [hunk 1]
```diff
@@ -1841,7 +1841,7 @@ static int32_t esp_task_get_max_priority(void)
 
 static void *esp_malloc(uint32_t size)
 {
-  return malloc(size);
+  return kmm_malloc(size);
```
Comment (Ouss4) [inline arch/xtensa/src/esp32/esp32_wifi_adapter.c:1844]: Could please also change the `_free` member of the `g_wifi_osi_funcs` structure from `free` to `kmm_free`.

There are actually a few other violation of the sort in this file, especially around files management.  We had an internal PR that addressed all of those but it has stalled for other reasons.
Will fix more in the upcoming days.

Comment (xiaoxiang781216) [reply arch/xtensa/src/esp32/esp32_wifi_adapter.c:1844]: @Ouss4 Done, I left the pthread mutex since it need more work to replace.

Comment (Ouss4) [reply arch/xtensa/src/esp32/esp32_wifi_adapter.c:1844]: Thanks

Comment (Ouss4) [reply arch/xtensa/src/esp32/esp32_wifi_adapter.c:1844]: @xiaoxiang781216 did you forget to push?

Comment (xiaoxiang781216) [reply arch/xtensa/src/esp32/esp32_wifi_adapter.c:1844]: Ohh, I forgot push my change, it should OK now.

**Review by Ouss4 — APPROVED** [review]
LGTM!
Thanks @xiaoxiang781216, it looks like you fixed everything for the esp32 adapter.

@jerpelea @davids5 please merge if you don't have any concerns.

**Pull request merged**